impeller/display_list/skia_conversions.cc

```diff
@@ -121,7 +121,9 @@ Path ToPath(const SkPath& path) {
   }
   builder.SetConvexity(path.isConvex() ? Convexity::kConvex
                                        : Convexity::kUnknown);
-  return builder.TakePath(fill_type);
+  auto result = builder.TakePath(fill_type);
+  result.SetOriginalGenerationId(path.getGenerationID());
```

Comment:
Where does this generation ID come from?

Reply:
It's assigned to path on request from a global counter. Any time the page is mutated it is reset. That way if path has same generationId it is guaranteed to be same instance unmodified.